packages/vite/src/node/plugins/worker.ts

```diff
@@ -73,8 +82,15 @@ export function webWorkerPlugin(config: ResolvedConfig): Plugin {
         url = injectQuery(url, WorkerFileId)
       }
 
+      const workerConstructor =
+        query.sharedworker != null ? 'SharedWorker' : 'Worker'
+      const { worker, inline, sharedworker, ...optionsFromQuery } = query
+      const workerOptions = { ...optionsFromQuery, type: 'module' }
```

Comment:
Looks good but I am not sure about passing all options from the query to be able to specify `name` and `credentials` for workers
```
./my-shared-worker?sharedworker&name=shared
```
I think it is safer to pick up the ones we want to support. There is also a discussion to have about naming. I am leaning towards `name` and `credentials` as in your PR.

Since `name` is intended for debugging, and `Worker` also has these options but are not yet supported, I think that we should add this feature in a separate PR. What do you think?

Reply:
Yeah, I remove other worker options except `type`.